OZON.ru.js

```diff
@@ -1,563 +1,346 @@
-		nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-		var valueXPathRes = doc.evaluate(
-		valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-		if (!IsAudioBook(nameXPathRes, valueXPathRes)) return "book";
-	} else {
-		itemTypeXPath = '//ul[@class="navLine"]/li[2]/a';
-		itemTypeDOMNode =
-		doc.evaluate(itemTypeXPath, doc, null,
-		XPathResult.ANY_TYPE, null).iterateNext();
-		if (itemTypeDOMNode) itemType = itemTypeDOMNode.textContent;
-		else return "";
-		if (itemType == "Цифровые книги") return "book";
-	}
-}
+	var hasIsbn = ZU.xpathText(doc, '//*[@itemprop="isbn"]') !== null;
+	var hasContent = ZU.xpath(doc, '//div[@class="bContentBlock"]') !== null;
 
-function IsAudioBook(nameXPathRes, valueXPathRes) {
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Формат аудиокниги") return true;
-	}
-	return false;
+	if (hasContent && hasIsbn) return 'book';
 }
 
-function MergeObjects(obj1, obj2) {
-	var result = new Object();
-	for (var attrName in obj1) {
-		result[attrName] = obj1[attrName];
-	}
-	for (var attrName in obj2) {
-		result[attrName] = obj2[attrName];
-	}
-	return result;
-}
+function doWeb(doc, url) {
+	var book = new Zotero.Item('book');
 
-function CleanText(text) {
-	var result = text.replace(/^\s*|\s*$/g, '');
-	result = result.replace(/\s+/g, ' ');
-	return result;
-}
+	var content = ZU.xpath(doc, '//div[@class="bContentBlock"]')[0];
 
-function CleanCreator(creatorStr) {
-	return CleanText(creatorStr);
-}
+	// basic info
+	book.title = ZU.xpathText(content, './h1[@itemprop="name"]');
+	book.url = url;
 
-function GetCreators(creatorsStr, creatorsCategory) {
-	var result = new Array();
-	var creator;
-	if (creatorsStr.match(",")) {
-		// http://www.ozon.ru/context/detail/id/7341440/
-		var authors = creatorsStr.split(",");
-		for (var i in authors) {
-			creator = CleanCreator(authors[i]);
-			creator = Zotero.Utilities.cleanAuthor(creator, creatorsCategory);
-			result.push(creator);
-		}
-	} else {
-		// http://www.ozon.ru/context/detail/id/7586040/
-		creator = CleanCreator(creatorsStr);
-		creator = Zotero.Utilities.cleanAuthor(creator, creatorsCategory);
-		result.push(creator);
-	}
-	return result;
-}
+	book.publisher = ZU.xpathText(content, './/*[@itemprop="publisher"]/a');
+	book.numPages = ZU.xpathText(content, './/*[@itemprop="numberOfPages"]');
 
-function GetLanguage(languageStr) {
-	switch (languageStr) {
-	case "Русский":
-		return "ru";
-	case "Английский":
-		return "en";
-	default:
-		return languageStr;
-	}
-}
 
-function GetISBN(isbnDateStr) {
-	return isbnDateStr.replace(/;\s.*$/, "");
-}
+	var isbn = ZU.xpathText(content, './/*[@itemprop="isbn"]');
+	if (isbn !== null) {
+		book.ISBN = ZU.cleanISBN(isbn);
+		book.date = /\d{4}(?=\s\u0433\.)/.exec(isbn)[0];
```

Comment:
Sorry if I mislead you, but I did only mean to ask to escape unicode in character ranges and where a character may be ambiguous (e.g. a single Cyrillic capital `A`). So most of what you had was ok. Should also make a note of the unescaped character.